packages/sdk/src/sync-agent/timeline/models/timeline-types.ts

```diff
@@ -0,0 +1,396 @@
+    ENCRYPTING = 'encrypting',
+    /** The event is in the process of being sent. */
+    SENDING = 'sending',
+    /** The event is in a queue waiting to be sent. */
+    QUEUED = 'queued',
+    /** The event has been sent to the server, but we have not yet received the echo. */
+    SENT = 'sent',
+    /** The event was cancelled before it was successfully sent. */
+    CANCELLED = 'cancelled',
+    /** We received this event */
+    RECEIVED = 'received',
+}
+
+export interface TimelineEvent {
+    eventId: string
+    localEventId?: string // if this event was created locally and appended before addEvent, this will be set
+    eventNum: bigint
+    latestEventId: string // if a message was edited or deleted, this will be set to the latest event id
+    latestEventNum: bigint // if a message was edited or deleted, this will be set to the latest event id
+    status: EventStatus
+    createdAtEpochMs: number // created at times are generated client side, do not trust them
+    updatedAtEpochMs?: number // updated at times are generated client side, do not trust them
+    content: TimelineEvent_OneOf | undefined // TODO: would be great to have this non optional
+    fallbackContent: string
+    isEncrypting: boolean // local only, isLocalPending should also be true
+    // isLocalPending: boolean /// true if we're waiting for the event to get sent back from the server
+    // isSendFailed: boolean
+    confirmedEventNum?: bigint
+    confirmedInBlockNum?: bigint
+    threadParentId?: string
+    replyParentId?: string
+    reactionParentId?: string
+    isMentioned: boolean
+    isRedacted: boolean
+    sender: {
+        id: string
+    }
+    sessionId?: string
+}
+
+/// a timeline event should have one or none of the following fields set
+export type TimelineEvent_OneOf =
+    | MiniblockHeaderEvent
+    // | NoticeEvent // TODO: understand
+    | ReactionEvent
+    | FulfillmentEvent
+    | KeySolicitationEvent
+    | PinEvent
+    | RedactedEvent
+    | RedactionActionEvent
+    // | RoomCanonicalAliasEvent // TODO: understand
+    // | RoomEncryptionEvent // TODO: understand
+    // | RoomAvatarEvent // TODO: understand
+    | RoomCreateEvent
+    | RoomMessageEncryptedEvent
+    // | RoomMessageMissingEvent // TODO: understand
+    | RoomMemberEvent // TODO: understand
+    | RoomMessageEvent
+    // | RoomNameEvent // TODO: understand
+    | RoomPropertiesEvent // TODO: understand (can we change the name to ChannelPropertiesEvent?)
+    // | RoomTopicEvent // TODO: understand
+    | ChannelCreateEvent // NOTE: prev SpaceChild
+    // | SpaceUpdateAutojoinEvent  // TODO: implement
+    // | SpaceUpdateHideUserJoinLeavesEvent // TODO: implement
+    // | SpaceImageEvent // TODO: implement
+    // | SpaceParentEvent // TODO: understand
+    | SpaceUsernameEvent
+    | SpaceDisplayNameEvent
+    | SpaceEnsAddressEvent
+    | SpaceNftEvent
+    | RoomMessageEncryptedRefEvent
+    | UnpinEvent
+
+export enum RiverEvent {
```

Comment:
Any suggestion to name this?

Reply:
TimelineEventType?
some of these are unused. can you go through on harmony and cmd+shift+f on them and delete them if we're not using it?

Reply:
https://github.com/HereNotThere/harmony/pull/8264